includes/reports/overview/list-table/class-sensei-reports-overview-list-table-abstract.php

```diff
@@ -114,21 +119,51 @@ public function prepare_items() {
 			$args['search'] = esc_html( $search );
 		}
 
-		$filters           = array_merge( $args, $this->get_additional_filters() );
-		$this->items       = $this->data_provider->get_items( $filters );
+		$this->total_items = $this->data_provider->get_last_total_items();
+		return array_merge( $args, $this->get_additional_filters() );
+	}
+
+	/**
+	 * Prepare the table with different parameters, pagination, columns and table elements
+	 *
+	 * @return void
+	 * @since  1.7.0
+	 */
+	public function prepare_items() {
+		$filter_args       = $this->get_filter_args();
+		$this->items       = $this->data_provider->get_items( $filter_args );
 		$this->total_items = $this->data_provider->get_last_total_items();
 
 		$total_items = $this->total_items;
-		$total_pages = ceil( $total_items / $per_page );
+		$total_pages = ceil( $total_items / $filter_args['number'] );
 		$this->set_pagination_args(
 			array(
 				'total_items' => $total_items,
 				'total_pages' => $total_pages,
-				'per_page'    => $per_page,
+				'per_page'    => $filter_args['number'],
 			)
 		);
 	}
 
+	/**
+	 * Get all the item ids.
+	 *
+	 * @return array The post ids.
+	 */
+	protected function get_all_item_ids() {
+			$result = $this->data_provider->get_items(
+				array_merge(
+					$this->get_filter_args(),
+					[
+						'number'         => -1,
+						'fields'         => 'ids',
+						'posts_per_page' => -1,
```

Comment:
I think the `get_items` method uses only the `number` key. Is `posts_per_page` actually needed?

Reply:
Yeah, you are right https://github.com/Automattic/sensei/pull/5091/commits/21c46e7d5febf44591b2ae9db69a625cc92dc7be